core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadWriteLDAPUserStoreManager.java

```diff
@@ -1347,7 +1347,7 @@ public void doUpdateRoleListOfUser(String userName, String[] deletedRoles, Strin
                         if (isExistingRole(deletedRole)) {
                             roleSearchFilter = "(&" + searchFilter.replace("?",
                                     escapeSpecialCharactersForFilter(deletedRole)) + "(" + membershipAttribute + "=" +
-                                    userNameDN + "))";
+                                    escapeSpecialCharactersForFilter(userNameDN) + "))";
```

Comment:
This fix should be added to `UniqueIDReadWriteLDAPUserStoreManager.java` class as well.

Reply:
Fixed